chore(benchmarks): add development cluster setup

npepinpe opened the pull request:
## Description

Adds instructions on how to setup a development cluster from scratch. I ran into this issue when migrating from Helm 2 to 3, where I had no idea how the cluster had been set up and we didn't keep track of everything in the repository.

This also now accurately reflects how the cluster is setup, and additionally improves things a little bit:

- Grafana now has a persistent disk
- Grafana now automatically provisions the Zeebe dashboard
- Option to install Istio
- Option to install linkerd

## Pull Request Checklist

- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

Zelldon commented on benchmarks/cluster/prometheus-operator-values.yml:
```diff
@@ -0,0 +1,49 @@
+  enabled: false
+
+grafana:
+  adminPassword: camunda
```
Thats why I havent added it the last time to the benchmark folder :sweat_smile:

npepinpe replied:
Yeah I thought about it, but it's not like it's a good password anyway. You'd still need to know the private IP of our Grafana instance. We could also always make it private again and you'd need to port-forward it, meaning you'd need access to the cluster, so really even having a password is kind of pointless :sweat_smile:

npepinpe commented:
Hm, I figured this was only partially docs, since some of it is the infrastructure itself (e.g. the manifest), but definitely the README is docs. :thinking:

npepinpe commented:
I moved the cluster/ stuff to the docs/setup/ folder, and removed docs/kubernetes/ as I'm not sure it's really useful anymore? It seemed to be redundant information that was already present in other documents (mostly the main README), and we don't use kustomize anymore anyway, so it was quite outdated.

npepinpe commented on benchmarks/docs/setup/README.md:
````diff
@@ -1,29 +1,90 @@
+To do so, run:
+
+```sh
+helm install metrics stable/prometheus-operator --atomic
````
```suggestion
helm install metrics stable/prometheus-operator --atomic -f prometheus-operator-values.yml
```

npepinpe commented:
bors r+
